File: 229-majority-element-ii/majority-element-ii.java
class Solution {

    public List<Integer> majorityElement(int[] nums) {
      //   int powerof2s = 0;
      //   int res = 0;
      // List<Integer> ll = new ArrayList<>();
      //   int range = nums.length / 3;
      //   for (int bits = 0; bits < 32; bits++) {
      //       int ones = 0, zeroes = 0;
      //       for (int element : nums) {
      //           if ((element & (1 << bits)) == 0) zeroes++; else ones++;
      //       }
      //       if (ones > range) {
      //           res += Math.pow(2, powerof2s);
      //       }
      //       powerof2s++;
      //   }
      HashMap<Integer,Integer> hm = new HashMap<>();
      List<Integer>ll = new ArrayList<>();
      for(int n:nums)
      {
        if(hm.containsKey(n))
        {
          hm.replace(n,hm.get(n)+1);
          
        }
        else hm.put(n,1);
      }
      for(int n:hm.keySet())
      {
        if(hm.get(n)>(nums.length/3))ll.add(n);
      }
      return ll;
      
   
     
    }
}

File: 229-majority-element-ii/majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {

        int maj1 = -1  , maj2 = -1;
        int cnt1 = 0 , cnt2 = 0;
        vector<int> res;

        for(int n:nums)
        {
            if(n == maj1)
            {
                cnt1++;

            }else if(n == maj2)
            {
                cnt2++;

            }else if(cnt1 == 0)
            {
                maj1=n;
                cnt1++;

            }else if(cnt2 == 0)
            {
                maj2 = n;
                cnt2++;

            }else {
                cnt1--;
                cnt2--;


            }
        }

        cnt1 = 0;
        cnt2 = 0;

        for(int n:nums)
        {
            
            if(n==maj1)cnt1++;
            else if(n==maj2)cnt2++;

        }


        if(cnt1>floor(nums.size()/3))
        {
            res.push_back(maj1);
        }

        if(cnt2>floor(nums.size()/3))
        {
            res.push_back(maj2);
        }


        return res;
        
    }
};